server/player/player.go

```diff
@@ -1058,3 +1058,19 @@ func (d Data) Save(db *sql.DB) error {
 func (d *Data) UpdateGuildInfo() {
 	d.Send(packetGuildInfo(0, "[Admins]", 0))
 }
+
+// DamagePlayer reduces character HP based on damage
+func (d *Data) DamagePlayer(damage int16, show bool) {
+	if damage < -1 {
+		return
+	}
+	newHP := d.hp - damage
+
+	if newHP <= -1 {
+		d.hp = 0
+	} else {
+		d.hp = newHP
+	}
+
+	d.Send(packetPlayerStatChange(show, constant.HpID, int32(d.hp)))
```

Comment:
The packet function has the bool at the front listed as unknown, Have you tested to see what it does?

Reply:
I did test that, but I did not see any difference for melee or bump damage when it was provided true or false. I had forgotten to remove it from the function. Removed and amended